server/gateway.go

```diff
@@ -1550,8 +1546,6 @@ func (c *client) processGatewayRUnsub(arg []byte) error {
 	ei, _ := c.gw.outsim.Load(accName)
 	if ei != nil {
 		e = ei.(*outsie)
-		e.Lock()
-		defer e.Unlock()
 		// If there is an entry, for plain sub we need
 		// to know if we should store the sub
 		useSl = queue != nil || e.mode != modeOptimistic
```

Comment:
e.mode ok to access here?

Reply:
nope.. let me revert the move of the lock/unlock (that I made at the last moment before pushing)..

Reply:
Can be RLock I think..

Reply:
no, because I update the `e` structure here..